### ISSUE TITLE
Large Table rendering

### ISSUE THREAD

**Issue by sgratzl** [girder/viime#261]
so far the cleanup table renders the whole dataset. Assuming that 1000 columns and 100 rows should be supported. That leads to way too many DOM elements. \

https://github.com/Akryum/vue-virtual-scroller could be used as an alternative. since we have more columns than rows, we could use it in the horizontal mode

**Comment by subdavis** [comment]
I agree.  I was also interested in using https://www.npmjs.com/package/canvas-datagrid because it could limit elements along both X and Y, and because I think table and row highlighting would work better. 

Couple of questions to answer about vue-virtual-scroller:

* How would you deal with all the custom styling we need to do?
* can it do fixed header and fixed row?  It would be really nice to have these things.

**Comment by jeffbaumes** [comment]
@sgratzl please do a deep technical dive into the capabilities of the vue table library. What I'm thinking is an independent repo hosted on github pages that renders a large (maybe 200 row, 2,000 column) table with fixed column headers (maybe row headers too?) and has dynamic styling like we need in viime. For example, click a button to change a certain row/column background color, color some random cells to indicate missing values. An exact match of the viime look is not needed, but a proof of concept that demonstrates it would be possible to do. I think rendering tables is something we need a good solution for, so a deeper investigation will be worth the time exploring.

**Comment by sgratzl** [comment]
current version: 

repo: https://github.com/sgratzl/vue-scroller-demo/
deployed: https://kind-kilby-e0d616.netlify.com/?cols=2000&rows=200

**Comment by subdavis** [comment]
Wow, that was super quick.  This looks really good.  Thanks for adding the fixed header and row.

How would re-rendering work?  Would it be possible to add a feature to click on a row or column header cell and highlight it?  After that, I think it would be reasonable to talk about integrating this into ~metabulo~ viime

**Comment by waxlamp** [comment]
Also, how flexible is the styling for things like grid lines, etc.? If we can use this but adopt the styling we've developed for Viime so far, then we're looking at the best of all possible worlds here.

**Comment by sgratzl** [comment]
![image](https://user-images.githubusercontent.com/4129778/65613656-62eeb900-df84-11e9-8b4b-9e16373ed399.png)

click on column header / row header to toggle highlight

**Comment by waxlamp** [comment]
This looks great! @subdavis what do you think?

In the meantime, could you summarize a bit how this works? I notice that if I rapidly scroll to the right, it feels a slight bit janky--is this from how the component is managing "pages" of data?

**Comment by sgratzl** [comment]
In the end it renders the whole table column wise. Since we have way more columns than rows the paging / viewport dependent is applied in column direction.

vue-virtual-list has a horizontal scroll listeners and renders just the columns that are currently visible. In addition, one can configure a buffer (200px by default) which extra columns to both sides should be rendered. Yes when you scroll fast it has to recreate all the columns but you cannot do much about it

**Comment by waxlamp** [comment]
Could you address @subdavis's question about re-rendering?

This seems to be just about at the point where you should prototype it in Viime, aiming for feature parity with the status quo.

**Comment by sgratzl** [comment]
> Could you address @subdavis's question about re-rendering?

>>  Would it be possible to add a feature to click on a row or column header cell and highlight it?

when clicking on the column or row header, a vue property will be set and a re-rendering takes place. It just works without thinking to much about it

**Comment by subdavis** [comment]
Sounds good to me.  Go for it.

**Comment by waxlamp** [comment]
@sgratzl thanks for putting this PoC together and answering our questions.

Please go ahead and prototype an integration into Viime. Since this is a substantial technology change, I'd like if you could open a draft PR as soon as you have an MVP for this so that the team can provide ongoing feedback and suggestions, instead of in bulk at the end of the process.
